src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -140,6 +145,22 @@
 )
 
 
+@bigquery_dialect.segment(replace=True)
+class SelectClauseModifierSegment(BaseSegment):
```

Comment:
Initially, the BigQuery dialect was supporting `SELECT AS STRUCT` by replacing `SelectClauseSegment` (see below). In working on this issue, I noticed that `AS STRUCT` is a general modifier that can be used on any `SELECT` statement. Implementing it this way is simpler and more generic -- doing it this way will likely avoid some future issues involving combinations of "all the things".

Reply:
sounds good :+1: